stub/include/neurala/video/VideoSourceStatus.h

```diff
@@ -29,60 +29,46 @@
 namespace neurala
 {
 /**
- * @brief Enum that describes the status of a VideoSource
+ * @brief Type that describes the status of a VideoSource
  */
-enum class VideoSourceStatus
+struct VideoSourceStatus final
 {
 	/// Operation was successful.
-	success,
+	static constexpr VideoSourceStatus success() { return {0}; }
 	/// Fetching frames timed out.  This is considered an error.
-	timeout,
+	static constexpr VideoSourceStatus timeout() { return {1}; }
 	/*
 	 * Internal buffer overflow (e.g. frames not fetched in time).
 	 * Currently buffered frames may still be fetched, however,
 	 * so this may or may not be an error condition depending on
 	 * whether skipped frames are allowed.
 	 */
-	overflow,
+	static constexpr VideoSourceStatus overflow() { return {2}; }
 	/*
 	 * All other errors.  (E.x. Requested frame number is outside valid range, no frame data
 	 * available, VideoInput connection was closed...)
 	 */
-	error,
+	static constexpr VideoSourceStatus error() { return {3}; }
 	/// Input format is not understood by the converters
-	pixelFormatNotSupported,
+	static constexpr VideoSourceStatus pixelFormatNotSupported() { return {4}; }
 	/// Information provided is not correct
-	invalidParameter,
+	static constexpr VideoSourceStatus invalidParameter() { return {5}; }
 	/// Functionality is not implemented
-	notImplemented,
+	static constexpr VideoSourceStatus notImplemented() { return {6}; }
 	/// Functionality not supported
-	unsupportedAction,
+	static constexpr VideoSourceStatus unsupportedAction() { return {7}; }
 	/// Unknown state.
-	unknown
+	static constexpr VideoSourceStatus unknown() { return {8}; }
```

Comment:
I hesitate to change this to the value 2 (in line with the other enums) in case there are side effects. If we're sure it's safe, we could try it.

Reply:
Do it.